x/nexus/keeper/general_message.go

```diff
@@ -84,32 +93,48 @@ func (k Keeper) SetMessageExecuted(ctx sdk.Context, messageID exported.MessageID
 
 // SetMessageFailed sets the general message as failed
 func (k Keeper) SetMessageFailed(ctx sdk.Context, messageID exported.MessageID) error {
-	m, found := k.GetMessage(ctx, messageID)
+	m, found := k.GetMessageAnyStatus(ctx, messageID)
 	if !found {
 		return fmt.Errorf("general message %s not found", messageID.String())
 	}
 
-	if !m.Is(exported.Sent) {
-		return fmt.Errorf("general message is not sent")
+	if !(m.Is(exported.Sent) || m.Is(exported.Approved)) {
+		return fmt.Errorf("general message is not sent or approved")
 	}
 
 	m.Status = exported.Failed
 
 	return k.setMessage(ctx, m)
 }
 
-// GetMessage returns the general message by ID
-func (k Keeper) GetMessage(ctx sdk.Context, messageID exported.MessageID) (m exported.GeneralMessage, found bool) {
-	return m, k.getStore(ctx).GetNew(getMessageKey(messageID), &m)
+// GetMessageAnyStatus returns the general message matching the given ID
+func (k Keeper) GetMessageAnyStatus(ctx sdk.Context, messageID exported.MessageID) (m exported.GeneralMessage, found bool) {
+	return k.GetMessageWithStatus(ctx, messageID, []exported.GeneralMessage_Status{})
+}
+
+// GetMessageWithStatus returns the general message matching the given ID and one of the passed in statuses
+func (k Keeper) GetMessageWithStatus(ctx sdk.Context, messageID exported.MessageID, statuses []exported.GeneralMessage_Status) (m exported.GeneralMessage, found bool) {
```

Comment:
```suggestion
func (k Keeper) GetMessage(ctx sdk.Context, messageID exported.MessageID, statuses ...exported.GeneralMessage_Status) (m exported.GeneralMessage, found bool) {
```

Reply:
Removed status from the key, so this is gone now